### PR TITLE
Ns search endpoint v2

### PR DESCRIPTION
**Pull request by PRocha0503**

### CODE REVIEW

File: src/listings/search/index.ts [hunk 1]
```diff
@@ -7,53 +7,45 @@ import internalServerErrorResponse from 'helpers/buildServerErrorResponse'
 
 export const handler: APIGatewayProxyHandler = async (event): Promise<APIGatewayProxyResult> => {
   try {
-    const query = event.pathParameters?.query
+    const { query, from, pageSize, ...exactFilters } = event.queryStringParameters || {}
 
-    if (!query) {
-      return badRequestResponse('Missing query parameter')
+    console.log('query', query)
```
Comment (StephanGuingor) [inline src/listings/search/index.ts:12]: can we remove the console log

File: src/listings/search/index.ts [hunk 2]
```diff
@@ -7,53 +7,45 @@ import internalServerErrorResponse from 'helpers/buildServerErrorResponse'
...
+
+    // Build the filter clauses for exact matches
+    const filterClauses =
+      exactFilters &&
+      Object.keys(exactFilters).map(field => ({
+        term: {
+          [field]: exactFilters[field],
```
Comment (StephanGuingor) [inline src/listings/search/index.ts:19]: would range filters be a part of second version?

File: src/listings/search/index.ts [hunk 3]
```diff
@@ -7,53 +7,45 @@ import internalServerErrorResponse from 'helpers/buildServerErrorResponse'
 
 export const handler: APIGatewayProxyHandler = async (event): Promise<APIGatewayProxyResult> => {
   try {
-    const query = event.pathParameters?.query
+    const { query, from, pageSize, ...exactFilters } = event.queryStringParameters || {}
```
Comment (StephanGuingor) [inline src/listings/search/index.ts:10]: we should validate pageSize, have a max size such as 50.

File: src/listings/search/index.ts [hunk 4]
```diff
@@ -7,53 +7,45 @@ import internalServerErrorResponse from 'helpers/buildServerErrorResponse'
...
-          },
-        },
+        from: from,
+        _source: ['year', 'region', 'id', 'price', 'full_name', 'colors'],
+        size: pageSize,
+        query: searchQuery,
       },
     })
-
     const listings = searchResults.hits.hits.map((hit: any) => ({
```
Comment (StephanGuingor) [inline src/listings/search/index.ts:49]: could we also return the total amount of hits? i think it is contained in the searchResults

**Review by StephanGuingor — CHANGES_REQUESTED** [review]

File: src/listings/search/index.ts (hunk 4 above)
Comment (PRocha0503) [reply src/listings/search/index.ts:49]: No

**Review by DiegoAraque21 — APPROVED** [review]

**Review by StephanGuingor — APPROVED** [review]

**Pull request merged**